### ISSUE TITLE
[WCAG 2.2 Draft Feedback] 3.2.7 Hidden Controls

### ISSUE THREAD

**Issue by dshoukry** [w3c/wcag#1455]
“Success Criterion 3.2.7 Hidden Controls (Level AA): Controls needed to progress or complete a process are visible at the time they are needed without requiring pointer hover or keyboard focus, or a mechanism is available to make them persistently visible.”

Most of our comments/proposals are requests to: clarify if the requirement for controls being “visible" also means that the control must not just appear on the screen, but also meet all other SCs that would affect visibility, like: [meaningful sequence](https://www.w3.org/TR/WCAG21/#meaningful-sequence), [sensory characteristics](https://www.w3.org/TR/WCAG21/#sensory-characteristics), and everything under [distinguishable](https://www.w3.org/TR/WCAG21/#distinguishable), and also how the visible requirement applies to items not directly visible, but within composite controls with visible entry points, such as an overflow menu or combobox, revisions to success examples, add a section for failure examples, and update links that 404.

Please find detailed specifics covered in our [3.2.7 Hidden Controls (Level AA) Google Doc](https://docs.google.com/document/d/1tt5238ELP3HJKRXNUc9DoPndR9aZ2Mgkm_dnL9xlgfA/edit#heading=h.rv4cak38r610).

**Comment by dshoukry** [comment]
As requested, I've just replaced the Google Doc link with a version with expanded access (didn't make any content changes). Sorry for the access issues with the previous link.

**Comment by alastc** [comment]
Hi @dshoukry,

Thank you for the comments. I'll address some below, but based on this and other issues the[latest version](https://w3c.github.io/wcag/understanding/visible-controls.html) uses a different approach which focuses on the information for that control being available is visible.

For comments which have become outdated by the update, I'll just put a small comment-replies in the document.

Draft response for the other comments:
<hr>

> 1.4.11 Non-text Contrast (Level AA) has a note (copy-pasted at the end of this comment) about the difficulty of determining a solution to make the contrast of inactive/disabled controls adequate for as many users as possible. Is that work still in progress?

We did consider an update in WCAG 2.2 for that, but until a feasible solution has been found, we don't have the basis for creating a criteria for that.

> There has been an increasing pattern of mobile apps offering accessibility settings to control the persistence of controls.  Can the text clarify if providing explicit preferences offering control over persistence is considered an acceptable technique?
> 
> Similarly, controls that time-out now have some platform-level hooks that allow users to express their preferred time allowances for disappearing controls. Can the text clarify whether or not this counts as an approved exception?

The SC text now has an exception for: "A mechanism is available to make the information persistently visible;" which would apply to both scenarios.

Also, PR #1617 was created for some of the editorial suggestions.

**Comment by alastc** [comment]
The response above was agreed by the group.
